src/AbstractPolyhedron_functions.jl

```diff
@@ -211,3 +212,113 @@ function remove_redundant_constraints(constraints::AbstractVector{LinearConstrai
         return EmptySet{N}()
     end
 end
+
+"""
+    linear_map(M::AbstractMatrix{N},
+               P::AbstractPolyhedron{N};
+               check_invertibility::Bool=true,
+               cond_tol::Number=DEFAULT_COND_TOL,
+               use_inv::Bool=!issparse(M)
+               ) where {N<:Real}
+
+Concrete linear map of a polyhedron in constraint representation.
+
+### Input
+
+- `M` -- matrix
+- `P` -- polyhedron in constraint representation
+- `check_invertibility` -- (optional, deault: `true`) check if the linear map is
+                           invertible, in which case this function uses the matrix
+                           inverse; if the invertibility check fails, or if
+                           this flag is set to `false`, use the vertex representation
+                           to compute the linear map (see below for details)
+- `cond_tol` -- (optional) tolerance of matrix condition (used to check whether
+                the matrix is invertible)
+- `use_inv`  -- (optional, default: `false` if `M` is sparse and `true`
+                otherwise) whether to compute the full left division through
+                `inv(M)`, or to use the left division for each vector; see below 
+
+### Output
+
+The result type depends on the input and the algorithm that was used:
+
+- If the vertex-based approach was used and the dimension of `P` is `2`, the output
+  is a `VPolygon`, otherwise a `VPolytope`.
+- If the input set is an `AbstractPolytope` , then the output is an `HPolytope`.
+- If the input set is an `AbstractPolygon` , then the output is an `HPolygon`.
+- The types of `HalfSpace`, `Hyperplane` and `Line` are preserved.
```

Comment:
Merge the last two lines:
``The types of `HalfSpace`, `Hyperplane`, `Line`, and `AbstractHPolygon` are preserved.``

Reply:
Done, and i also splitted the paragraphs into two separate ones, depending on the algorithm that was used.